backend/src/Edna/ExperimentReader/Types.hs

```diff
@@ -0,0 +1,61 @@
+module Edna.ExperimentReader.Types
+  ( Parameter (..)
+  , PlateUnit (..)
+  , ParameterType (..)
+  , Signal (..)
+  , CellType (..)
+  , PointYX (..)
+  ) where
+
+import Universum
+
+import qualified GHC.Show as S
+
+import Codec.Xlsx (CellValue(..))
+import Fmt (Buildable(..), pretty, (+|), (|+))
+
+newtype PointYX = PointYX (Int, Int)
+  deriving stock (Show, Eq)
+
+data ParameterType = Target | Compound
+
+data Parameter = Parameter
+  { pName :: Text
+  , pIndexes :: (Int, Int)
+  } deriving stock Show
+
+data PlateUnit = PlateUnit
+  { tuTargets :: [Parameter]
+  , tuCompounds :: [Parameter]
+  }
+
+data Signal = Signal
+  { sValue :: Double
+  , sOutlier :: Bool
+  }
+
+data CellType :: Type -> Type where
+ CText  :: CellType Text
+ CDouble :: CellType Double
+ CSignal :: CellType Signal
+
+instance Buildable PointYX where
+  build (PointYX (y, x)) = "{row=" +| y |+ ", column=" +| x |+ "}"
+
+instance Buildable (CellType a) where
+  build CText = "text"
+  build CDouble = "double"
+  build CSignal = "signal"
+
+instance Show (CellType a) where
+  show = pretty
+
+instance Eq (CellType a) where
+  (==) a b = (pretty a :: Text) == pretty b
+
+instance Buildable CellValue where
```

Comment:
Note that it raises the following warning:
```
[ 5 of 12] Compiling Edna.ExperimentReader.Types ( src/Edna/ExperimentReader/Types.hs, /home/gromak/Work/biocad/edna/backend/dist-newstyle/build/x86_64-linux/ghc-8.10.3/edna-0/noopt/build/Edna/ExperimentReader/Types.dyn_o )

src/Edna/ExperimentReader/Types.hs:36:1: warning: [-Worphans]
    Orphan instance: instance Buildable CellValue
    To avoid this
        move the instance declaration to the module of the class or of the type, or
        wrap the type with a newtype and declare the instance on the new type.
   |
36 | instance Buildable CellValue where
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

If this instance is needed only for `cellSatisfy`, it may be better to simply remove it, see my comment around that function. That being said, I am more or less ok with having this instance, so if you want to keep it, let's keep it and just suppress `-Worphans` in this particular file.

Reply:
Not only for `cellSatisfy`, also for error showing